uktrade/data-hub-frontend
# Add api for add pipeline feature

## Description of change

This PR adds the pipelines apis and functionality for adding a company and checking if the company is already been added to a pipeline. 
The PR includes 

- Task for posting data to the endpoint
- Task for and logic for checking if a company has been added to a pipeline
- Controller for setting a flash message
- useEffect for redirecting to the dashboard once redux store has success data

This is part  of yellow team's mission work for this quarter to build functionality that allows users to add companies to three defined statuses - lead, in progress and win - in a single pipeline. This will work in a similar way to the existing company list feature.

The url for this page is not linked anywhere on the site.

## Test instructions

When you navigate to /companies/<Company:ID>/pipeline. You will see a form asking which pipeline you wish to add the company to. Once added you'll be redirected back to the dashboard with a success flash message. Navigate back to /companies/<Company:ID>/pipeline and you should see the form and a messaging saying COMPANY_NAME is already in your pipeline. 

## Screenshots
### Already in Pipeline 
### Before
![image](https://user-images.githubusercontent.com/5575331/81573243-1f727f00-939c-11ea-8b50-02e3f42c1814.png)
![image](https://user-images.githubusercontent.com/5575331/81573331-3b762080-939c-11ea-85ca-c1825662d14d.png)

### Not in Pipeline 
_Add a screenshot_
![image](https://user-images.githubusercontent.com/5575331/81576703-77ab8000-93a0-11ea-8e38-210806613eb1.png)


### Added to Pipeline 
![image](https://user-images.githubusercontent.com/5575331/81576592-5a76b180-93a0-11ea-9bfe-2951a4b79825.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
